enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -3838,6 +3841,306 @@ class AdjointGenerator
+        if ((xoverwrite && !gutils->isConstantValue(call.getArgOperand(3))) ||
+            (yoverwrite && !gutils->isConstantValue(call.getArgOperand(1)))) {
+          if (Mode == DerivativeMode::ReverseModeGradient &&
+              (!gutils->isConstantValue(call.getArgOperand(1)) ||
+               !gutils->isConstantValue(call.getArgOperand(3)))) {
+            cacheval = Builder2.CreatePHI(cachetype, 0);
+          }
+          cacheval = gutils->cacheForReverse(Builder2, cacheval,
+                                             getIndex(&call, CacheType::Tape));
+          if (!gutils->isConstantValue(call.getArgOperand(1)) &&
+              !gutils->isConstantValue(call.getArgOperand(3)) && xoverwrite &&
+              yoverwrite) {
+            structarg1 = BuilderZ.CreateExtractValue(cacheval, 0);
+            structarg2 = BuilderZ.CreateExtractValue(cacheval, 1);
+          } else if (!gutils->isConstantValue(call.getArgOperand(3)) &&
+                     xoverwrite)
+            structarg1 = cacheval;
+          else if (!gutils->isConstantValue(call.getArgOperand(1)) &&
+                   yoverwrite)
+            structarg2 = cacheval;
+        }
+        CallInst *firstdcall, *seconddcall;
+        if (!gutils->isConstantValue(call.getArgOperand(3))) {
+          assert(!isa<Constant>(orig->getArgOperand(3)));
+          Value *D1 = gutils->invertPointerM(orig->getArgOperand(3), Builder2);
+          SmallVector<Value *, 6> args1 = {
+              lookup(gutils->getNewFromOriginal(orig->getArgOperand(0)),
+                     Builder2),
+              diffe(orig, Builder2),
+              structarg1,
+              Builder2.getInt32(1),
+              D1,
+              Builder2.getInt32(1)};
+          firstdcall = Builder2.CreateCall(daxpycall, args1);
+        }
+        if (!gutils->isConstantValue(call.getArgOperand(1))) {
+          assert(!isa<Constant>(orig->getArgOperand(1)));
+          Value *D2 = gutils->invertPointerM(orig->getArgOperand(1), Builder2);
+          SmallVector<Value *, 6> args2 = {
+              lookup(gutils->getNewFromOriginal(orig->getArgOperand(0)),
+                     Builder2),
+              diffe(orig, Builder2),
+              structarg2,
+              Builder2.getInt32(1),
+              D2,
+              Builder2.getInt32(1)};
+          seconddcall = Builder2.CreateCall(daxpycall, args2);
+        }
+        setDiffe(orig, Constant::getNullValue(orig->getType()), Builder2);
+        if (!gutils->isConstantValue(call.getArgOperand(3)) && xoverwrite)
+          CallInst::CreateFree(structarg1, firstdcall->getNextNode());
+        if (!gutils->isConstantValue(call.getArgOperand(1)) && yoverwrite)
+          CallInst::CreateFree(structarg2, seconddcall->getNextNode());
+      }
+      return;
+    }
+
+    if (funcName == "cblas_sdot" && called->isDeclaration()) {
+      Type *castvals[2] = {call.getArgOperand(1)->getType(),
```

Comment:
I'd have this be handled in the same code as above with a if funcName == "cblas_ddot" || funcName == "cblas_sdot". Then when it differs (per type), you can do a check whether it's ddot or sdot